File: diretorioprincipal/questao3.go
//atividade 06/09/24
//aluno: João Pedro Araújo Inácio
//questao3

package main

import "fmt"

type ContaBancaria struct {
    titular string
    saldo   float64
}

func NovaContaBancaria(titular string, saldoInicial float64) *ContaBancaria {
    return &ContaBancaria{titular: titular, saldo: saldoInicial}
}

func (c *ContaBancaria) Depositar(valor float64) {
    if valor > 0 {
        c.saldo += valor
        fmt.Printf("valor depositado: %.2f, saldo atual: %.2f\n", valor, c.saldo)
    } else {
        fmt.Println("Valor de deposito invalido.")
    }
}

func (c *ContaBancaria) Sacar(valor float64) {
    if valor > 0 && valor <= c.saldo {
        c.saldo -= valor
        fmt.Printf("Sacado: %.2f, saldo atual: %.2f\n", valor, c.saldo)
    } else {
        fmt.Println("Saldo insuficiente ou valor invalido.")
    }
}

func (c *ContaBancaria) GetSaldo() float64 {
    return c.saldo
}

func (c *ContaBancaria) GetTitular() string {
    return c.titular
}

func main() {
    conta := NovaContaBancaria("João Pedro", 1000)
    conta.Depositar(200)
    conta.Sacar(500)
    fmt.Printf("Saldo final: %.2f\n", conta.GetSaldo())
}

File: diretorioprincipal/questao3.py
#atividade 06/09/24
#aluno: João Pedro Araújo Inácio
#questao3

class ContaBancaria:
    def __init__(self, titular, saldo_inicial=0):
        self.__titular = titular
        self.__saldo = saldo_inicial

    def depositar(self, valor):
        if valor > 0:
            self.__saldo += valor
            print(f'valor depositado: {valor}, saldo atualmente: {self.__saldo}')
        else:
            print("valor de deposito invalido.")

    def sacar(self, valor):
        if 0 < valor <= self.__saldo:
            self.__saldo -= valor
            print(f'Sacado: {valor}, Saldo atual: {self.__saldo}')
        else:
            print("Saldo insuficiente ou valor invalido.")

    def get_saldo(self):
        return self.__saldo

    def get_titular(self):
        return self.__titular

conta = ContaBancaria('Joao Pedro', 1000)
conta.depositar(200)
conta.sacar(500)
print("Saldo final:", conta.get_saldo())